test/Microsoft.AspNet.Mvc.Core.Test/ControllerActionInvokerTest.cs

```diff
@@ -1203,6 +1203,647 @@ private struct SampleStruct
         }
 
         [Fact]
+        public async Task InvokeAction_InvokesAsyncResourceFilter()
+        {
+            // Arrange
+            var resourceFilter = new Mock<IAsyncResourceFilter>(MockBehavior.Strict);
+            resourceFilter
+                .Setup(f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()))
+                .Returns<ResourceExecutingContext, ResourceExecutionDelegate>(async (c, next) =>
+                {
+                    await next();
+                })
+                .Verifiable();
+
+            var invoker = CreateInvoker(new IFilter[] { resourceFilter.Object });
+
+            // Act
+            await invoker.InvokeAsync();
+
+            // Assert
+            resourceFilter.Verify(
+                f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()),
+                Times.Once());
+        }
+
+        [Fact]
+        public async Task InvokeAction_InvokesResourceFilter()
+        {
+            // Arrange
+            var resourceFilter = new Mock<IResourceFilter>(MockBehavior.Strict);
+            resourceFilter
+                .Setup(f => f.OnResourceExecuting(It.IsAny<ResourceExecutingContext>()))
+                .Verifiable();
+            resourceFilter
+                .Setup(f => f.OnResourceExecuted(It.IsAny<ResourceExecutedContext>()))
+                .Verifiable();
+
+            var invoker = CreateInvoker(new IFilter[] { resourceFilter.Object });
+
+            // Act
+            await invoker.InvokeAsync();
+
+            // Assert
+            resourceFilter.Verify(
+                f => f.OnResourceExecuted(It.IsAny<ResourceExecutedContext>()),
+                Times.Once());
+            resourceFilter.Verify(
+                f => f.OnResourceExecuted(It.IsAny<ResourceExecutedContext>()),
+                Times.Once());
+
+        }
+
+        [Fact]
+        public async Task InvokeAction_InvokesAsyncResourceFilter_WithActionResult_FromAction()
+        {
+            // Arrange
+            ResourceExecutedContext context = null;
+            var resourceFilter = new Mock<IAsyncResourceFilter>(MockBehavior.Strict);
+            resourceFilter
+                .Setup(f => f.OnResourceExecutionAsync(It.IsAny<ResourceExecutingContext>(), It.IsAny<ResourceExecutionDelegate>()))
+                .Returns<ResourceExecutingContext, ResourceExecutionDelegate>(async (c, next) =>
+                {
+                    context = await next();
+                })
+                .Verifiable();
+
+            var invoker = CreateInvoker(new IFilter[] { resourceFilter.Object });
```

Comment:
sure, there's plenty more like this though, and it doesn't seem useful to try and clean them as it's just sugar in the API.